tests/testthat/test-linear_pool.R

```diff
@@ -453,13 +453,130 @@ test_that("samples only collected and re-indexed for simplest case", {
     linear_pool(
       weights = NULL,
       task_id_cols = c("target_date", "target", "horizon", "location"),
+      comp_unit_cols = c("target", "location"),
       n_output_samples = NULL
     )
 
   expect_equal(actual_outputs, expected_outputs)
 })
 
 
+test_that("ensemble of samples correctly drawn for compound task ID sets", {
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
```

Comment:
Per comments on the code above, let's add a test where different models submit for different subsets of locations.

Reply:
Implemented! (https://github.com/hubverse-org/hubEnsembles/pull/147/commits/2c89864f3e16eb47b5e54a3749b5292251ee4a20)